core/network/conn.go

```diff
@@ -37,11 +38,11 @@ type Conn interface {
 // ConnectionState holds information about the connection.
 type ConnectionState struct {
 	// The stream multiplexer used on this connection (if any). For example: /yamux/1.0.0
-	StreamMultiplexer string
+	StreamMultiplexer protocol.ID
 	// The security protocol used on this connection (if any). For example: /tls/1.0.0
-	Security string
+	Security protocol.ID
 	// the transport used on this connection. For example: tcp
-	Transport string
+	Transport protocol.ID
```

Comment:
The transport is not really a `protocol.ID`. Can we leave this as a `string`?

Reply:
fixed. For my understanding, protocol.ID is used for any protocol we use *over* the transport, right?